src/fparser/two/utils.py

```diff
@@ -843,59 +858,92 @@ def tostr(self):
 
 
 class BracketBase(Base):
-    """
-::
-    <bracket-base> = <left-bracket-base> <something> <right-bracket>
-    """
+    '''
+    bracket-base is left-bracket something right-bracket.
+
+    This class is able to cope with nested brackets as long as they
+    are correctly nested. Brackets in strings are ignored.
+
+    The 'something' can be specified as being optional.
+
+    '''
     @staticmethod
     def match(brackets, cls, string, require_cls=True):
-        ''' The generic match method for all types of bracketed
-        expressions '''
-        bracket_len = len(brackets)//2
-        left = brackets[:bracket_len]
-        right = brackets[-bracket_len:]
-
-        if not (string.startswith(left) and string.endswith(right)):
-            return
+        '''A generic match method for all types of bracketed
+        expressions.
+
+        :param str brackets: the format of the left and right brackets \
+        provided as a string, for example '()'
+        :param cls: the class to match the content within the brackets \
+        :type cls: subclass of :py:class:`fparser.two.utils.Base`
+        :param str string: the content to match
+        :param bool require_cls: whether the class and associated \
+        content is mandatory (True) or optional (False). The default \
+        is True.
+        :return: None if there is no match, otherwise a tuple with the \
+        first and third entries being strings containing the left and \
+        right brackets respectively and the second entry being either \
+        None or an instance of the class provided as the second \
+        argument (cls).
+        :rtype: 'NoneType', ( `str`, `NoneType`, `str`) or ( `str`, \
+        `cls`, `str` )
 
+        '''
+        if not cls and require_cls:
+            return None
+        if not string:
+            return None
+        string_strip = string.strip()
+        if not brackets:
+            return None
+        brackets_nospc = brackets.replace(' ', '')
+        if not brackets_nospc:
+            return None
+        if len(brackets_nospc) % 2 == 1:
+            # LHS and RHS bracketing must be the same size
+            return None
+        bracket_len = len(brackets_nospc)//2
+        left = brackets_nospc[:bracket_len]
+        right = brackets_nospc[-bracket_len:]
+        if len(string_strip) < bracket_len:
```

Comment:
Should that be `< 2*bracket_len` or have I missed something?

Reply:
You are quite right. The test I added to test_bracket_base.py was not failing at this location. The reason I didn't notice in terms of coverage is that an existing Fortran2003 test was failing here.